src/query/plan/preprocess.hpp

```diff
@@ -125,6 +125,88 @@ struct Expansion {
   NodeAtom *node2 = nullptr;
 };
 
+struct FilterMatching;
+
+enum class PatternFilterType { EXISTS };
```

Comment:
Does it make sense to have an enum class with a single value?

Reply:
I just added it for later extension, so we do no assumptions when creating pattern filters now. does it make sense? :')

Reply:
i use it just in a if-else followed by an assert(false

Reply:
If you use enums, then use `switch` without `default` branches: reason: if you extend the enum but don't handle it in an switch, then the compiler will scream about it and the code won't compile. With if+assert you are pushing the same check to runtime.

Reply:
Whether to keep the enum or not: what other types of pattern filter can we have?

Reply:
for me this is okay, in case we expand usage for later with something else